soroban-token-sdk/src/event.rs

```diff
@@ -0,0 +1,49 @@
+use soroban_sdk::{symbol_short, Address, Env, Symbol};
+
+pub struct Events {
+    env: Env,
+}
+
+impl Events {
+    #[inline(always)]
+    pub fn new(env: &Env) -> Events {
+        Events { env: env.clone() }
+    }
+
+    pub fn approve(&self, from: Address, to: Address, amount: i128, expiration_ledger: u32) {
+        let topics = (Symbol::new(&self.env, "approve"), from, to);
+        self.env
+            .events()
+            .publish(topics, (amount, expiration_ledger));
+    }
+
+    pub fn transfer(&self, from: Address, to: Address, amount: i128) {
+        let topics = (symbol_short!("transfer"), from, to);
+        self.env.events().publish(topics, amount);
+    }
+
+    pub fn mint(&self, admin: Address, to: Address, amount: i128) {
+        let topics = (symbol_short!("mint"), admin, to);
+        self.env.events().publish(topics, amount);
+    }
+
+    pub fn clawback(&self, admin: Address, from: Address, amount: i128) {
+        let topics = (symbol_short!("clawback"), admin, from);
+        self.env.events().publish(topics, amount);
+    }
+
+    pub fn set_authorized(&self, admin: Address, id: Address, authorize: bool) {
+        let topics = (Symbol::new(&self.env, "set_authorized"), admin, id);
+        self.env.events().publish(topics, authorize);
+    }
+
+    pub fn set_admin(&self, admin: Address, new_admin: Address) {
```

Comment:
This now contains a mix of events that correspond to the generic token interface and `StellarAssetAdminInterface`. So maybe we should consider separating this as well? I don't have a strong opinion though.

Reply:
Splitting up the token interface makes sense because you can choose which subset to implement, but in this case these are just helpers for you to call. I don't think we need to split them up.